lib/SIL/SILVerifier.cpp

```diff
@@ -1498,8 +1498,12 @@ class SILVerifier : public SILVerifierBase<SILVerifier> {
     require(origTy, "The original function must have a function type");
     require(!origTy->isDifferentiable(),
             "The original function must not be @differentiable");
-    if (F.getModule().getStage() == SILStage::Canonical ||
-        dfi->hasDerivativeFunctions()) {
+    // Skip lowered SIL: LoadableByAddress changes parameter/result conventions.
+    // TODO: Check that derivative function types match excluding
```

Comment:
Note: robust verification here should be aware of the lowered SIL stage and check that address-lowered SIL function types match.

In the meantime, we could use a custom function for recursively checking SIL function type equality, ignoring parameter/result conventions. `SILFunctionType::isABICompatibleWith` is not suitable because it checks parameter/result conventions and returns an opaque `ABICompatibilityCheckResult`.